source/backend/api/Services/LeaseService.cs

```diff
@@ -471,5 +489,67 @@ private void AppendNewItemsToChecklist(PimsLease lease, ref List<PimsLeaseCheckl
                 }
             }
         }
+
+        private void ValidateRenewalDates(PimsLease lease, ICollection<PimsLeaseRenewal> renewals)
+        {
+            if (lease.LeaseStatusTypeCode != PimsLeaseStatusTypes.ACTIVE)
+            {
+                return;
+            }
+
+            List<Tuple<DateTime, DateTime>> renewalDates = new();
+
+            foreach (var renewal in renewals)
+            {
+                if (renewal.IsExercised == true)
+                {
+                    if (renewal.CommencementDt.HasValue && renewal.ExpiryDt.HasValue)
+                    {
+                        renewalDates.Add(new Tuple<DateTime, DateTime>(renewal.CommencementDt.Value, renewal.ExpiryDt.Value));
+                    }
+                    else
+                    {
+                        throw new BusinessRuleViolationException("Excercised renewals must have a commencement date and expiry date");
+                    }
+                }
+            }
+
+            // Sort agreement dates/renewals by start date.
+            renewalDates.Sort((a, b) => DateTime.Compare(a.Item1, b.Item1));
+
+            DateTime currentEndDate;
+
+            if (lease.OrigStartDate.HasValue && lease.OrigExpiryDate.HasValue)
+            {
+                var agreementStart = lease.OrigStartDate.Value;
+                var agreementEnd = lease.OrigExpiryDate.Value;
+                currentEndDate = agreementEnd;
+                if (DateTime.Compare(agreementEnd, agreementStart) <= 0)
+                {
+                    throw new BusinessRuleViolationException("The lease commencement date must be before its expiry date");
+                }
+            }
+            else
+            {
+                throw new BusinessRuleViolationException("Active leases must have commencement and expiry dates");
+            }
+
+            for (int i = 0; i < renewalDates.Count; i++)
+            {
+                var startDate = renewalDates[i].Item1;
+                var endDate = renewalDates[i].Item2;
+
+                if (DateTime.Compare(endDate, startDate) <= 0)
+                {
+                    throw new BusinessRuleViolationException("The expiry date of your renewal should be later than its commencement date");
+                }
+
+                if (DateTime.Compare(currentEndDate, startDate) >= 0)
+                {
+                    throw new BusinessRuleViolationException("The commencement date of your renewal should be later than the previous expiry date (agreement or renewal)");
```

Comment:
I feel like this satisfies the ac in abstract, but not on a 1-1 basis. Make sure there is a not on the story for Praveen so he doesn't fail this.

Reply:
Also, should the user be able to allow colliding renewal dates when the renewal is not exercised? currently I can.

Reply:
I had a conversation with julian about this. A non-exercised renewal is functionally a draft, so no validation happens for those